lib/elixir/lib/kernel/typespec.ex

```diff
@@ -373,6 +373,7 @@ defmodule Kernel.Typespec do
   defp builtin_type?(:nonempty_charlist, 0), do: true
   defp builtin_type?(:keyword, 0), do: true
   defp builtin_type?(:keyword, 1), do: true
+  defp builtin_type?(:var, 0), do: true
```

Comment:
This is a breaking change.

It seems small, but I understand if you prefer to emit a warning in v1.8.0 that can be turned into an error in future Elixir versions.

Reply:
On the positive side, we explicitly document that we may introduce new functions in kernel, new modules or types and those may cause conflict which could be breaking. So I think it would be worth giving this a try anyway.